### ISSUE TITLE
Chinese writing instead of the japanese one

### ISSUE THREAD

**Issue by Quezako** [chanind/hanzi-writer-data-jp#7]
Hello,

Thanks a lot for this great library, it is very useful!

But unfortunately, I've already encountered many N5 and N4 kanji with a different writing, like for 終.
![Hanzi Writer - Google Chrome 01_10_2022 02_56_58](https://user-images.githubusercontent.com/98868/193330044-a5230ec9-20fc-441d-973c-9c278c958336.jpg)
The bottom left is different. This is a problem, Because I use this library to practice my kanji writing, and I've already lost points on my kanji tests because of that.

The trickier one was for 祭, which is almost the same, except for the stroke 5!
![298](https://user-images.githubusercontent.com/98868/193329925-5ac59818-ed40-45dc-9a8c-b6847d67ba99.gif)
![_](https://user-images.githubusercontent.com/98868/193329930-31150c00-8a9f-4a66-b843-a508a0d930c9.gif)

Is there a way to get real japanese writing data somewhere?

**Comment by chanind** [comment]
The data for Japanese comes from https://github.com/parsimonhi/animCJK, although it hasn't been updated in a few years. Is this character correct in that repo? If so I can try to pull in the most recent version from there and that should hopefully fix the issue. Sorry for any problems caused by this!

**Comment by Quezako** [comment]
No, unfortunately, it is the same kanji in CJK.
I was wondering if there is a way to convert a good version in .svg to the used .json format?
Because, there are good in KanjiVG
https://kanjivg.tagaini.net/viewer.html?kanji=%E7%B5%82
The SVG : https://kanjivg.tagaini.net/kanjivg/kanji/07d42.svg

What kind of format are the json files? is it documented somewhere?

**Comment by Quezako** [comment]
I've found a writing check tool with using kanjivg on kanshudo:
https://www.kanshudo.com/play/kanjidraw?&min_ufn=1&max_ufn=1&gm=high
But I don't think that it's open source and it seems that it is only for kanji, not jukugo.

**Comment by chanind** [comment]
Is the issue just with the stroke order, or the actual outlines of the character strokes? If it's just stroke order you can rearrange the strokes / medians in JSON, but if there's issues with the character outline that's trickier. the general idea is that you need both the SVG outline of each stroke, and also the `medians` which is an animation line indicating how that stroke should be animated. The SVG format mimics what's in [makemeahanzi](https://github.com/skishore/makemeahanzi), but I'm not sure how hard it is to turn any SVG into a path that's compatible.

**Comment by Quezako** [comment]
Yes, it is the strokes who are different and not correct in a japanese school.
Ok, so I see that for instance, for 終 :
The json file is: https://cdn.jsdelivr.net/npm/hanzi-writer-data@2.0/%E7%B5%82.json
The data is the same in makemeahanzi svg: view-source:https://raw.githubusercontent.com/skishore/makemeahanzi/master/svgs/32066.svg
And here is the kanjivg svg: view-source:https://kanjivg.tagaini.net/kanjivg/kanji/07d42.svg

I see in the kanjivg file, that there are 2 parts:
StrokePaths:
```
<g id="kvg:StrokePaths_07d42" style="fill:none;stroke:#000000;stroke-width:3;stroke-linecap:round;stroke-linejoin:round;">
<g id="kvg:07d42" kvg:element="終">
	<g id="kvg:07d42-g1" kvg:element="糸" kvg:position="left" kvg:radical="general">
		<path id="kvg:07d42-s1" kvg:type="㇜" d="M28.15,14.5c0.31,1.25,0.41,2.46-0.16,3.78c-2.61,6.1-8.19,15.05-11.07,18.19c-1.17,1.28-1.3,2.66,0,3.11c3.3,1.16,5.84,3.07,8.64,5.35"/>
```

And StrokeNumbers:
```
<g id="kvg:StrokeNumbers_07d42" style="font-size:8;fill:#808080">
	<text transform="matrix(1 0 0 1 21.75 15.13)">1</text>
```
As they are used in the kanshudo link I gave you early to detect each stroke order, start and end.
But I don't know at which point it is similar to the hanzi writer writing detection?
Did you code the stroke detection and correction system, or is ti coming from another source?
Thank you for your help.

**Comment by yaozhang-this** [comment]
As a Chinese who also studies Japanese, I can confirm there are many such differences and animCJK's data sources are simply not up for such task. Each differences between the Chinese style and Japanese style would need to be fixed for all related kanjis, for example, here is another stroke that Japanese schools teach that differs from what modern Chinese writing, the first stroke is a dot in Chinese but a vertical stroke in Japanese, https://kakijun.jp/gif-mini/0665200-m.gif And it requires the font SVG that was used as the basis of outline to be changed, and that can't be done without switching to a completely different set of SVG for Japanese renderings(in 宋体/明体Songti/Mingti(明朝体Minchotai in japanese) that will look very much different from the style used in Chinese Hanzis (楷书Kaisu (or 楷書Kaisho in Japanese). At that point it is probably easier to just use KanjiVG for that.

**Comment by parsimonhi** [comment]
Hello, I am the author of animCJK.

AnimCJK and Hanzi-writer are in kaisho style.

1) 終: in Japanese kaisho style, the stroke order of the bottom left of 終 is left, middle, right. This is an exception, not an error, one of the few exceptions between the kaisho style (楷書体, regular style) vs the kyōkasho style (教科書体, textbook style). Do not panic, this is not so difficult to take this into account. And it is the same for all characters when the radical　糹 (糸)　 is "on the left". And the radical is written as 糹in this case (not as 糸). See the 4th image of https://kakijun.jp/shotaigifnew/11185200.gif or https://moji.tekkai.com/gif-DFks/E7B582.gif

When 糸 is on the bottom of a character, the order remains middle, left, right in kaisho style as in kyōkasho style.

2) 祭: in Japanese kaisho style, the 5th stroke of 祭 doesn't cross the sixth stroke. No error in animCJK/hanzi-writer. See the 4th image of https://kakijun.jp/shotaigifnew/11165200.gif or https://moji.tekkai.com/gif-DFks/E7A5AD.gif

3) 字: in Japanese kaisho style, the 1st stroke is oblique (from top left to bottom right). No error in animCJK/hanzi-writer. See 4th image of https://kakijun.jp/shotaigifnew/0665200.gif or https://moji.tekkai.com/gif-DFks/E5AD97.gif

4) Do not use Make-Me-A-Hanzi in Japanese. It is designed for Chinese, not for Japanese. There are several thousand of characters that are different in Chinese vs Japanese (glyph, stroke order, number of stroke). The appropriate adaptations were done "one by one" in animCJK, but may be there are still some errors in animCJK (nobody is perfect). In this case please open an issue on https://github.com/parsimonhi/animCJK (you are welcome).

In summary, it is not a problem of Chinese vs Japanese style or a problem in animCJK/hanzi-writer. It is just a question of style!

Note also that I recently updated animCJK and corrected some issues.

**Comment by Quezako** [comment]
Thank you for the detailed clarifications!
Unfortunately, if I write the kanji in kaisho style, I will get 0 point, since they want us to use the kyōkasho style while drawing kanji.
Maybe it could be interesting to have the kyōkasho style alternative dataset somewhere?
For the moment I'm using the Kanshudo tool and I've implemented it in my Anki kanji deck, for the drawing review.

**Comment by parsimonhi** [comment]
Excepting  糹 (糸) which is the main "problem", there are nearly no difference between the kaisho style and the kyōkasho style concerning the stroke order.

Note also that the Japanese manuscript style is closer to the kaisho style than to the kyōkasho style. So when you learn to "write", the kyōkasho style is not better.

And you will not avoid some "problems" anyway (for instance for characters containing 牙, 以, 令 etc. and characters that were changed recently by the Japanese).

**Comment by Quezako** [comment]
Ok, so I suppose that I can close this issue, since it is only a problem of writing style.
Thank you very much, I'll discuss this point with my teachers.

EDIT: I asked the teacher and she told me that the 糹way of writing is easier but not standard and more often used by old people in Japan. So, while it is interesting to know about it, I think that it may be misleading to learn this alternate way of writing, compared to the standard way taught in class. Kanji are already confusing enough! 😅
